website/pizzas/api/v2/admin/views.py

```diff
@@ -0,0 +1,118 @@
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import filters as framework_filters
+from rest_framework.generics import get_object_or_404
+
+import pizzas.api.v2.filters as normal_filters
+from pizzas.api.v2.admin.permissions import IsOrganiser
+from pizzas.api.v2.admin.serializers.food_event import FoodEventAdminSerializer
+from pizzas.api.v2.admin.serializers.order import FoodOrderAdminSerializer
+from pizzas.api.v2.admin.serializers.product import ProductAdminSerializer
+from pizzas.models import Product, FoodEvent, FoodOrder
+from thaliawebsite.api.v2.admin import (
+    AdminListAPIView,
+    AdminCreateAPIView,
+    AdminRetrieveAPIView,
+    AdminUpdateAPIView,
+    AdminDestroyAPIView,
+)
+
+
+class FoodEventAdminListView(AdminListAPIView, AdminCreateAPIView):
+    """Returns an overview of all food events."""
+
+    serializer_class = FoodEventAdminSerializer
+    queryset = FoodEvent.objects.all()
+    filter_backends = (
+        framework_filters.OrderingFilter,
+        normal_filters.FoodEventDateFilterBackend,
+    )
+    ordering_fields = ("start", "end")
+    permission_classes = [
+        IsAuthenticatedOrTokenHasScope,
+    ]
+    required_scopes = ["food:admin"]
+
+
+class FoodEventAdminDetailView(AdminRetrieveAPIView, AdminUpdateAPIView):
+    """Returns or updating one single food event."""
```

Comment:
```suggestion
    """Returns one single food event."""
```

The update part is left out in most other update views. Otherwise, I think 'updates' would be grammatically better.